tests/api/test_aws.py

```diff
@@ -303,39 +303,109 @@ def get_statements_by_sid(policy_document):
 )
 def test_grant_bucket_access(iam, users, resources):
     bucket_arn = 'arn:aws:s3:::test-bucket'
-    path_arns = [f'{bucket_arn}{resource}' for resource in resources]
+    path_arns_list = []
+    path_arns_object = []
+    if resources:
+        path_arns_list = [f'{bucket_arn}{resource}' for resource in resources]
+        path_arns_object = [f'{bucket_arn}{resource}/*' for resource in resources]
     user = users['normal_user']
     aws.create_user_role(user)
 
-    aws.grant_bucket_access(user.iam_role_name, bucket_arn, 'readonly', path_arns)
+    aws.grant_bucket_access(user.iam_role_name, bucket_arn, 'readonly', path_arns_list)
 
     policy = iam.RolePolicy(user.iam_role_name, 's3-access')
     statements = get_statements_by_sid(policy.policy_document)
 
-    if path_arns:
+    if path_arns_object:
+        assert set(path_arns_object) == set(statements['readonly']['Resource'])
         assert f'{bucket_arn}/*' not in statements['readonly']['Resource']
     else:
-        assert f'{bucket_arn}/*' in statements['readonly']['Resource']
+        assert set([f'{bucket_arn}/*']) == set(statements['readonly']['Resource'])
     # no readwrite statement because no readwrite access granted
     assert 'readwrite' not in statements
-    if path_arns:
-        assert set(path_arns) == set(statements['list']['Resource'])
-    else:
-        assert bucket_arn in statements['list']['Resource']
+    assert set([bucket_arn]) == set(statements['list']['Resource'])
 
     aws.grant_bucket_access(user.iam_role_name, f'{bucket_arn}-2', 'readonly')
     policy.reload()
     statements = get_statements_by_sid(policy.policy_document)
     expected_num_resources = 2
-    if path_arns:
-        expected_num_resources = len(path_arns) + 1
+    if path_arns_list:
+        expected_num_resources = len(path_arns_list) + 1
     assert len(statements['readonly']['Resource']) == expected_num_resources
 
 
 @pytest.mark.parametrize(
     'resources',
     [
-        ([],),
+        (['/foo/bar', '/foo/baz'])
+    ],
+    ids=[
+        'paths'
+    ]
+)
+def test_revoke_bucket_path_access(iam, users, resources):
+    bucket_arn = 'arn:aws:s3:::test-bucket'
+    path_arns = [f'{bucket_arn}{resource}' for resource in resources]
```

Comment:
I suppose it doesn't matter but you may want to add a `/` between the `bucket_arn` and the `resource` in case this test fail as the output may be misleading.

Reply:
Maybe I'm not understanding, but would that not cause the test to fail incorrectly with the specified resources?

Reply:
You're probably right, TBH I can't remember much since that comment but by quick look at it it seems like the slash are already above - if I'll remember what I meant I'll let you know.